### PR TITLE
13 implement test cases for showcase app

### PR DESCRIPTION
**Pull request by svdwyer**
fixes https://github.com/SolidLabResearch/solid-identity-manager/issues/13

### CODE REVIEW

**Review by pheyvaer — CHANGES_REQUESTED** [review]
The showcase app has both a `package-lock.json` and `yarn.lock`.

**Comment by svdwyer** [comment]
Having both npm and yarn lock files for the plugin obviously doesn't make sense. The plugin was set up with yarn, so I could simply remove the npm lock file.
But the extension on the other hand uses npm. Shall we all both projects and use npm for both?

File: showcase-app/tests/main.spec.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  await mainPage.loadPage();
+  await expect(mainPage.getPage().getByRole('button', {name: /Continue as Test Profile A/})).toBeVisible();
+
+  await popupPage.createProfile('Test Profile B', 'Test IPD B');
+
+  await mainPage.loadPage();
+  await expect(mainPage.getPage().getByRole('button', {name: /Continue as Test Profile B/})).toBeVisible();
+});
+
+test('Removing profile inside the extension deactivates the profile', async ({page, mainPage, popupPage}) => {
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply showcase-app/tests/main.spec.js:38]: I meant removing the profile in the extension should deactive that profile in the plugin.
So if the plugin has profile A active (with the "Continue as A" shown), that should not be the case anymore after the profile is removed in the extension

[the remark this answers is not in the document]
Comment (pheyvaer) [reply showcase-app/tests/main.spec.js:38]: In that case it's better to write exactly that. Now there is also the confusion between "extension" and "plugin".

**Review by pheyvaer — CHANGES_REQUESTED** [review]
Where are the instructions on how to run the tests?

**Review by pheyvaer — CHANGES_REQUESTED** [review]
When I start the tests I get this in my terminal:

```shell
)$ npm test

> solid-auth-clientside-showcase@0.0.0 test
> PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS=1 playwright test
[WebServer] Error:   Failed to scan for dependencies from entries:
  /Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/index.html
/Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/playwright-report/index.html

  The package "@esbuild/darwin-x64" could not be found, and is needed by esbuild.

If you are installing esbuild with npm, make sure that you don't specify the
"--no-optional" or "--omit=optional" flags. The "optionalDependencies" feature
of "package.json" is used by esbuild to install the correct binary executable
for your current platform.
    at generateBinPath (/Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/node_modules/esbuild/lib/main.js:1885:15)
    at esbuildCommandAndArgs (/Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/node_modules/esbuild/lib/main.js:1955:33)
    at ensureServiceIsRunning (/Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/node_modules/esbuild/lib/main.js:2119:25)
    at Object.context (/Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/node_modules/esbuild/lib/main.js:2012:33)
    at prepareEsbuildScanner (file:///Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:43097:26)
    at async file:///Users/pheyvaer/Development/solid/solid-identity-manager/showcase-app/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:43597:38
```

File: showcase-app/test-cases.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 This document provides an overview of the end-to-end tests for the plugin.
 
+### How to run the tests
+
+To run end-to-end tests, you must first have built the plugin (`npm run build`).
```
[the remark this answers is not in the document]
Comment (svdwyer) [reply showcase-app/test-cases.md:7]: it should be run in the project folder: showcase-app
npm will look for a package.json in the folder it is run in.

[the remark this answers is not in the document]
Comment (pheyvaer) [reply showcase-app/test-cases.md:7]: But won't that build the showcase app instead of the plugin then?

[the remark this answers is not in the document]
Comment (svdwyer) [reply showcase-app/test-cases.md:7]: I'm sorry, i meant it the other way: the build should happen in extension directory (project root)
I will amend the wording

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**